src/main/java/io/anserini/index/IndexCollection.java

```diff
@@ -799,8 +772,19 @@ public Counters run() throws IOException {
     LOG.info("Thread pool with " + numThreads + " threads initialized.");
 
     LOG.info("Initializing collection in " + collectionPath.toString());
-    final List segmentPaths = collection.getSegmentPaths();
+    List<?> segmentPaths = collection.getSegmentPaths();
+
+    // for C4 specifically we filter through segmentPaths to only take ones that we want based on file #
+    if (args.collectionClass.equals("C4Collection") && args.shardCount > 1) {
```

Comment:
I think it's clearer to do an outer `if (args.shardCount > 1)` and then an inner `if-else` checking the collection. If we add more collection-specific processing, it'll be cleaner.